### PR TITLE
Faster histograms

### PR DESCRIPTION
**Pull request by kadri-nizam**
I benchmarked a small part of the code base and found that the `np.histogram2d` call in `_sum_cell_values_channel` method is pretty slow. The method takes up almost 75% of the entire running time (22.2 s / 29.6 s). See the visualized profiling below with the relevant section highlighted in magenta.
image
Since our gridding process creates grids that are uniformly spaced, we can opt for libraries that specialize in histograms for uniform bin sizes. I added `fast-histogram` into project's dependency and substituted the histogram call. The method call now only takes 6.2% of the total running time (0.565 s / 9.12 s) resulting in a 3.25x speed up. Relevant section highlighted again in magenta below.
image

### CODE REVIEW

File: src/mpol/gridding.py [hunk 1]
```diff
@@ -246,15 +246,17 @@ def _sum_cell_values_channel(self, uu, vv, values=None):
             A 2D array of size ``(npix, npix)`` in ground format containing the summed cell quantities.
         """
 
-        result = np.histogram2d(
+        bin_range = self.coords.vis_ext[:2]
```
Comment (iancze) [inline src/mpol/gridding.py:249]: Could we make this code clearer to understand? For example, using `coords.v_bin_min`, and `coords.v_bin_max` and `coords.u_bin_min` etc..

**Review by iancze — APPROVED** [review]
Thanks, I think this is a nice improvement for a routine that does some heavy lifting (at least in the `DataAverager` and `DirtyImager` routines). It seems like the package is reasonably well maintained, otherwise I would be cautious about adding another core dependency outside of `numpy`.

File: src/mpol/gridding.py (hunk 1 above)
Comment (kadri-nizam) [reply src/mpol/gridding.py:249]: Sure thing! Somehow it didn't occur to me that those attributes were available. I'll push the changes shortly.

**Comment by kadri-nizam** [comment]
> otherwise I would be cautious about adding another core dependency outside of numpy.

I concur. I'll look into whether numpy has routines that are made specially for uniform bin sizes.

**Comment by kadri-nizam** [comment]
It looks like the author of `fast-histogram` also contributed to making [numpy's histogram faster](https://github.com/numpy/numpy/pull/6100). They made their own version that's fully implemented in C without the bells and whistles that numpy does.

**Comment by iancze** [comment]
Looks great, thanks! Given the fast-histogram author's commitment to various open source projects, I don't think it's a huge risk that that package will become outdated before MPoL does :) And, if it does, it should be easy enough to switch back to numpy-only.

**Pull request merged**